olive/systems/azureml/aml_workflow_runner.py

```diff
@@ -0,0 +1,51 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import argparse
+import json
+import shutil
+from pathlib import Path
+
+from olive.common.constants import WORKFLOW_ARTIFACTS, WORKFLOW_CONFIG
+from olive.common.utils import aml_runner_hf_login, copy_dir
+from olive.logging import set_verbosity_from_env
+from olive.workflows import run as olive_run
+
+
+def main(raw_args=None):
+    set_verbosity_from_env()
+
+    # login to hf if HF_LOGIN is set to True
+    aml_runner_hf_login()
+
+    parser = argparse.ArgumentParser("Olive workflow config")
+
+    parser.add_argument(f"--{WORKFLOW_CONFIG}", type=str, help="olive workflow config", required=True)
+    parser.add_argument(f"--{WORKFLOW_ARTIFACTS}", type=str, help="olive workflow artifacts", required=True)
+
+    olive_config_args = parser.parse_args(raw_args)
+
+    with open(olive_config_args.workflow_config) as f:
+        olive_config = json.load(f)
+
+    olive_run(olive_config, setup=True)
+    olive_run(olive_config)
+
+    workflow_artifacts_path = Path(olive_config_args.workflow_artifacts)
+
+    remote_cache_dir = workflow_artifacts_path / "cache"
+    remote_output_dir = workflow_artifacts_path / "output"
+
+    # Clean up the remote cache and output directories if they exist
+    if remote_cache_dir.exists():
```

Comment:
should we provide an option to reuse cache? For example, the workflow might fail at a later step but we want to reuse the previous steps?

Reply:
Most AML workspace compute should be provisioned as a new one for next job. I got some errors when a node in a cluster was not released and immediately used for next job so I added this. I don't think we would have any use case that reuse cache. What do you mean by reuse the previous steps?

Regarding to your question, this `cache` folder is not the Olive workflow `cache`. This is a remote path with same name.

Reply:
I see that the `workflow_artifacts_path` is mounted as a datastore folder. So, I thought this was used as the workflow cache. But I now see that the cache is copied after the workflow is run.

Reply:
this brings up another question. if a user runs two workflows concurrently or sequentially without changing the workflow id, wouldn't this cause the results of the run that completes first to disappear? or in an edge case, case a race condition on the directory save? 

Since the workflow cache is not persistent across run, the `workflow_id` doesn't seem useful if the user needs to specify it themselves. It made sense when the engine host was a remote vm. 
Maybe the windows_artifacts_path should be created using a unique_id for the specific config being run? Maybe a hash for the workflow config? Instead of `f"azureml://datastores/{self.config.datastores}/paths/{workflow_id}"`

Reply:
Yeah I was thinking cache is not useful for this remote run as we cannot use this cache at all. The reason I save cache folder as well to datastore is because: 1. Output models is in cache folder. Unless we copy output models out. 2. I (may) will add retrieve feature later. Then the cache can be downloaded locally for local use.

Reply:
What would be the local use of downloaded cache?

There is no use of local (i.e. on the remote workflow host) cache. Ideally, it would use cloud cache here from PR 1166; 
 which can be done as a follow up PR.

Reply:
For the same workflow-id, I was thinking a use case that a user runs workflow remotely, retrieve the cache and outputs to local. When next time he was trying to run the same workflow, Olive can directly go for local cache in this case. This is a different use case from could model cache.